### ISSUE TITLE
Usage of updatePointCloud leads to length_error in vector::reserve

### ISSUE THREAD

**Issue by MaxFleur** [PointCloudLibrary/pcl#4001]
**Description**
When using the viewer function `updatePointCloud` a `length_error` in `vector::reserve` is thrown out.

**Context**

I want to delete points out of a point cloud. To do that, I created a KeyEvent function that deletes the points using the BackSpace key.
The code to delete the newest point (which means the last index inside the point cloud) and update the cloud goes as follows:
`if(event.getKeySym () == "BackSpace" && event.keyDown()) {`
`if(!cloud.empty) {`
`// delete the "newest" point in the point cloud.`
`cloud->erase(cloud->end() - 1);`
`// Use any color you like to visualize the cloud`
`pcl::visualization::PointCloudColorHandlerCustom<pcl::PointXYZRGB> red(cloud, 235, 19, 19);`
`m_viewer->updatePointCloud(cloud, red, "cloud");`
`}`
`}`

**Expected behavior**

The user should be able to delete all points of the cloud until no points remain.

**Current Behavior**

If the point cloud contains more than three points and points are deleted until only two points remain, the following issue will be displayed if the user wants to delete a point again:
`terminate called after throwing an instance of 'std::length_error'
  what():  vector::reserve`
This error is only thrown if the function `updatePointCloud` is used. But if the function is used this error will be thrown while running the function `viewer->spinOnce(100);`

**To Reproduce**
1. Create a basic pcl project and create a random point cloud with at least three points.
2. Add a `KeyEvent` function using an arbitrary key to delete points using the function as mentioned above.
3. Delete the points until only two points remain.
4. Hit the arbitrary key again.

-> Now this error occures.
-> At first, I thought this issue might be caused due to the
`cloud->erase(cloud->end() - 1);`
because I thought it might be some bad code. But this is definitely not the issue. I tried to delete points using an ExtractIndices-function (as mentioned [here](docs.pointclouds.org/trunk/classpcl_1_1_extract_indices.html#add1af519a1a4d4d2665e07a942262aac) and the error occured nonetheless.
-> If the point cloud contains only two points, everything works fine, so no error is displayed. It only happens if the cloud contains three or more points.
-> I already created a stackoverflow entry for this issue (see [here](https://stackoverflow.com/questions/61505627/updatepointcloud-function-causes-stdlength-error-for-vectorreserve)) but there is now answer until now.

-> I have a picture of the full stacktrace after the error was thrown. I'll put it here:
![stacktrace](https://user-images.githubusercontent.com/15110943/80752550-f34d3780-8b2b-11ea-8429-6b8015793f34.png)

**Tested environment**

 - OS: ManjaroLinux 20.0
 - Compiler: GCC 9.3.0
 - PCL Version: 1.10

Let me know if you need some more information. Cheers!

**Comment by kunaltyagi** [comment]
This is not a MCVE.

Also the erase call looks suspiciously bad. Try `if (!vector.empty()) vec.erase(vec.crbegin());`

**Comment by MaxFleur** [comment]
> This is not a MCVE. Seems unrelated to PCL based on information provided 

Are there some more informations needed?

> Also the erase call looks suspiciously bad. Try `if (!vector.empty()) vec.erase(vec.crbegin());`

The erase call was used due to an administrator post from [here](http://www.pcl-users.org/Remove-Singe-Point-from-Pointcloud-td3304482.html). Also it's not possible to call `crbegin()`. It's only possible to do that for the points vector _within_ the cloud. Is that what you mean?

**Comment by kunaltyagi** [comment]
I didn't notice you were using the cloud. The usage appears correct, except that at `size() = 0`

Regarding MCVE, you example isn't minimum or easily verifiable. The issue also doesn't show that the problem is in the library or some code you wrote (in which case SO is the best place)

**Comment by MaxFleur** [comment]
Alright, I did some more testing and stripped down the issue. Hope it's more clear now.

**Comment by kunaltyagi** [comment]
> Delete the points until only two points remain

Possible issue: user-code: The delete then tries to delete 2 or 3 poins, trying to delete position `-1`

Closing in favor of the SO question since issue seems unrelated to PCL library code.

**Comment by MaxFleur** [comment]
The delete only deletes one point per function call and it is always the last point. This does work if the cloud only contains two points at the beginning. All ppints can be deleted. I've pointed that out- Also, I wrote that this problem occurs as well if an `ExtractIndices`-Function was used and that this issue only occures if the `updatePointCloud`-function is executed. 
Therefore, I don't really understand why this issue is closed now. :-\

**Comment by kunaltyagi** [comment]
* There is an issue in PCL `updatePointCloud` which appears if the new point cloud is larger than the older. This isn't the case here. You can try using `removePointCloud` + `addPointCloud` instead of update, but based on discussions, it will not help you.
* The other place where it could be PCL's fault is in `erase` function. But the check for empty (seems wrong to use  variable instead of a function-call) implies that can't be the reason.
* No other avenue is available because what you've provided isn't a MCVE. An MCVE is a self sustained piece of code

The last is the key point. Without that, based on first 2 points, I can't be sure that the issue isn't in user-code.

**Comment by MaxFleur** [comment]
Okay, I understand. Will remember it the next time. :-)
I found another issue on this page about this (listed [here](https://github.com/PointCloudLibrary/pcl/issues/3452).
So I tried it using the `removePointCloud` + `addPointCloud` as you and the other users suggested. Frankly, that did it. It works for now. Thank you!

Should I create an MCVE nontheless? I can provide a minimal example of code to give some more details, but the issue is already there so I#M not sure whther that's really necessary.
